packages/office-addin-unit-test/src/officeJSMock.ts

```diff
@@ -0,0 +1,150 @@
+import { usageDataObject } from "./defaults";
+
+export class OfficeJSMock {
+  constructor(object?: OfficeObject) {
+    this.properties = new Map<string, OfficeJSMock>();
+    this.loaded = false;
+    this.resetValue(undefined);
+    if (object) {
+      this.populate(object);
+    }
+  }
+
+  // Adds a function to OfficeJSMock
+  addMockFunction(methodName: string, functionality?: Function) {
+    this[methodName] = functionality ? functionality : function () {};
+  }
+
+  // Adds a OfficeJSMock to OfficeJSMock
+  addMockObject(objectName: string) {
+    const officeJSMock = new OfficeJSMock();
+    officeJSMock.isMockObject = true;
+    this.properties.set(objectName, officeJSMock);
+    this[objectName] = this.properties.get(objectName);
+  }
+
+  // load of Office.js API
+  load(property: string) {
+    this.loadMultipleProperties(property);
+  }
+
+  // Adds a property of any type to OfficeJSMock
+  setMock(propertyName: string, value: unknown) {
+    if (!this.properties.has(propertyName)) {
+      const officeJSMock = new OfficeJSMock();
+      officeJSMock.isMockObject = false;
+      this.properties.set(propertyName, officeJSMock);
+    }
+    this.properties.get(propertyName)?.resetValue(value);
+    this[propertyName] = this.properties.get(propertyName)?.value;
+  }
+
+  // sync of Office.js API
+  sync() {
+    this.properties.forEach((property: OfficeJSMock, key: string) => {
+      property.sync();
+      this.assignValue(key);
+    });
+    if (this.loaded) {
+      this.value = this.valueBeforeLoaded;
+    }
+  }
+
+  private assignValue(propertyName: string) {
+    if (this.properties.get(propertyName)?.isMockObject) {
+      this[propertyName] = this.properties.get(propertyName);
+    } else {
+      this[propertyName] = this.properties.get(propertyName)?.value;
+    }
+  }
+
+  private loadCalled() {
+    this.loaded = true;
+    this.value = `Error, context.sync() was not called`;
+  }
+
+  private loadMultipleProperties(property: string) {
```

Comment:
'loadMultipleProperties' but the argument is just one 'property'?

Reply:
The name is wrong, but the function is accepting multiple properties. Fixing it